shared/memorypool/memorypool.go

```diff
@@ -0,0 +1,27 @@
+package memorypool
```

Comment:
@nisdas this stuff seems unrelated to the PR, can it be made into a separate PR?

Reply:
alright, done #4896